src/helpers.ts

```diff
@@ -68,11 +68,22 @@ export async function requiredStatusChecksForBranch(
     })
   ).data
 
+  var checksFromBranchProtection = []
+
+  const rules = (
+    await octokit.rest.repos.getBranchRules({
+      ...github.context.repo,
+      branch: branchName,
+    })
+  ).data
+
+  const checksFromRules = rules.reduce((acc, rule) => acc.concat(rule?.parameters?.required_status_checks?? []), [])
```

Comment:
mmm, still getting `Base branch 'main' of pull request 36 is not sufficiently protected.`, and npm test still gives me this error
```
 FAIL  __tests__/helpers.test.ts
  ● Test suite failed to run

    src/helpers.ts:80:72 - error TS2339: Property 'parameters' does not exist on type '({ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }) | ({ type: "update"; parameters?: { ...; } | undefined; } & { ...; }) | ... 12 more ... | ({ ...; } & { ...; })'.
      Property 'parameters' does not exist on type '{ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }'.

    80   const checksFromRules = rules.reduce((acc, rule) => acc.concat(rule?.parameters?.required_status_checks?? []), [])
                                                                              ~~~~~~~~~~
    src/helpers.ts:83:5 - error TS2322: Type '{ context: string; app_id: number | null; }[] | undefined' is not assignable to type 'any[]'.
      Type 'undefined' is not assignable to type 'any[]'.

    83     checksFromBranchProtection = branch.protection.required_status_checks?.checks
           ~~~~~~~~~~~~~~~~~~~~~~~~~~

 FAIL  __tests__/input.test.ts
  ● Test suite failed to run

    src/helpers.ts:80:72 - error TS2339: Property 'parameters' does not exist on type '({ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }) | ({ type: "update"; parameters?: { ...; } | undefined; } & { ...; }) | ... 12 more ... | ({ ...; } & { ...; })'.
      Property 'parameters' does not exist on type '{ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }'.

    80   const checksFromRules = rules.reduce((acc, rule) => acc.concat(rule?.parameters?.required_status_checks?? []), [])
                                                                              ~~~~~~~~~~
    src/helpers.ts:83:5 - error TS2322: Type '{ context: string; app_id: number | null; }[] | undefined' is not assignable to type 'any[]'.
      Type 'undefined' is not assignable to type 'any[]'.

    83     checksFromBranchProtection = branch.protection.required_status_checks?.checks
           ~~~~~~~~~~~~~~~~~~~~~~~~~~

Running coverage on untested files...Failed to collect coverage from /Users/joshua.simon/Github/automerge/src/helpers.ts
ERROR: src/helpers.ts:80:72 - error TS2339: Property 'parameters' does not exist on type '({ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }) | ({ type: "update"; parameters?: { ...; } | undefined; } & { ...; }) | ... 12 more ... | ({ ...; } & { ...; })'.
  Property 'parameters' does not exist on type '{ type: "creation"; } & { ruleset_source_type?: "Repository" | "Organization" | undefined; ruleset_source?: string | undefined; ruleset_id?: number | undefined; }'.

80   const checksFromRules = rules.reduce((acc, rule) => acc.concat(rule?.parameters?.required_status_checks?? []), [])
                                                                          ~~~~~~~~~~
src/helpers.ts:83:5 - error TS2322: Type '{ context: string; app_id: number | null; }[] | undefined' is not assignable to type 'any[]'.
  Type 'undefined' is not assignable to type 'any[]'.

83     checksFromBranchProtection = branch.protection.required_status_checks?.checks
       ~~~~~~~~~~~~~~~~~~~~~~~~~~
STACK:
---------------------------------|---------|----------|---------|---------|-------------------
File                             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------------------------|---------|----------|---------|---------|-------------------
All files                        |       0 |        0 |       0 |       0 |
 automerge-action.ts             |       0 |        0 |       0 |       0 | 1-348
 codegen.ts                      |       0 |      100 |     100 |       0 | 3-15
 github-graphql-schema-loader.ts |       0 |      100 |     100 |       0 | 1-2
 index.ts                        |       0 |        0 |       0 |       0 | 1-53
 input.ts                        |       0 |        0 |       0 |       0 | 1-84
---------------------------------|---------|----------|---------|---------|-------------------
Test Suites: 2 failed, 2 total
Tests:       0 total
Snapshots:   0 total
Time:        5.843 s
```